src/JsonSchema/Constraints/UndefinedConstraint.php

```diff
@@ -66,7 +66,8 @@ public function validateTypes(&$value, $schema = null, JsonPointer $path, $i = n
         }
 
         // check object
-        if ($this->getTypeCheck()->isObject($value)) {
+        if (TypeCheck\LooseTypeCheck::isObject($value)) { // Fixes failing assoc tests for default values - currently investigating
+        //if ($this->getTypeCheck()->isObject($value)) {  // to find the root cause of this, noting all other assoc tests pass.
```

Comment:
Sorry for the delay.

I've stepped through your code, and the problem is that you are deferring to `testValidCases` from `testValidCasesUsingAssoc`, and you're passing along an array instead of an object.  If you look in BaseTestCase::testValidCasesUsingAssoc, you'll see that it prepares  `Validator`  by passing in `$checkMode`, and the default value of `$checkMode` is `CHECK_MODE_TYPE_CAST`, which will have no problem with associative arrays.

I see that you extended `VeryBaseTestCase` instead of `BaseTestCase`; presumably you did that because you wanted a different signature for `testValidCasesUsingAssoc` and `testValidCases`. That's okay, but make sure that you flesh out `testValidCasesUsingAssoc` with the proper boilerplate.